### ISSUE TITLE
SA1300 - '_Page_Views_Shared__Layout_cshtml'

### ISSUE THREAD

**Issue by huchim** [DotNetAnalyzers/StyleCopAnalyzers#2373]
Hi

I'm gettings this warning in my file `Views\Shared\_Layout.cshtml`

```bash
SA1300	Element '_Page_Views_Shared__Layout_cshtml' must begin with an uppercase letter.
```

I understand the rule, but i tried including this line on the top of the file with no success.

```html
// <auto-generated />
<!DOCTYPE html>   ...
```

(This isn't working, I guess because the view is compiled and thats is not thr first line)
```csharp
@{
// <auto-generated />
}
```

```bash
Microsoft .NET Core Shared Framework Host
  Version  : 1.1.0
  Build    : 928f77c4bc3f49d892459992fb6e1d5542cb5e86

Windows 7 x64
Visual Studio 2017
```

Thanks in advance... 😃

**Comment by tillig** [comment]
This appears to be a larger issue in general, at least for me. Using either version `1.0.2` or `1.1.0-beta001` of the StyleCop analyzers, all generated code based on Razor views is getting analyzed with StyleCop and failing all sorts of things - from documentation rules to code ordering rules.

It looks like the VS tooling folks have a related issue [here](https://github.com/aspnet/Tooling/issues/498) and it even mentions that StyleCop is getting kicked off against the generated code.

**Comment by tillig** [comment]
Note that, at least for me, many of the problems go away if I just _close the file in VS_. The doc warnings and StyleCop errors appear when the .cshtml file is open but not when it's closed, which is odd.

**Comment by sharwell** [comment]
@tillig Does it occur when you compile the project? By default, VS only analyzes open files. You can have it analyze all files by enabling Full Solution Analysis in the advanced C# options pages.

**Comment by tillig** [comment]
If I have a view open I see analysis errors. Rebuilding the solution doesn't get those errors to disappear.

![View Open](https://user-images.githubusercontent.com/1156571/26942212-0c154452-4c36-11e7-9dd9-58afe833458a.png)

If I close all the views and/or build the project from the command line, I don't see those warnings.

![View Closed](https://user-images.githubusercontent.com/1156571/26942253-348201dc-4c36-11e7-81ea-a7d4e4b0192a.png)

The challenge is just that if there actually _are_ warnings in there that I should pay attention to, there's no way to filter through and see what they are. It generates a lot of noise that shouldn't be there.

**Comment by jimmymain** [comment]
my feeling is that this noise is entirely a result of the fact that the code generated by razor simple doesn't include // <auto-generated/> on the first line.

[CS1591](https://github.com/aspnet/Tooling/issues/498) warnings are also generated.

**Comment by sharwell** [comment]
@jimmymain I'm not aware of a way to exclude things like that. We removed all of the exclusions code from version 1.1 in favor of allowing Roslyn to decide what to exclude (or not).

**Comment by jimmymain** [comment]
@sharwell, of course you're correct, that's exactly how it should work. I suspect just getting razor to add the `auto-generated` will actually solve all the problems, include 1591.

**Comment by sharwell** [comment]
@jimmymain The `<auto-generated/>` header does not affect CS1591. See dotnet/roslyn#12702 for more information.

**Comment by rynowak** [comment]
FYI we are tracking this feedback on Razor at aspnet/Razor#1449

**Comment by rynowak** [comment]
FYI for anyone watching this, this will be fixed in a later preview of Visual Studio 2017 15.3
